src/app/person/person.component.ts

```diff
@@ -9,38 +9,45 @@ import { PersonService } from "src/app/services/person.component.service";
   styleUrls: ["./person.component.scss"]
 })
 export class PersonComponent implements OnInit {
-  constructor(public router: Router, private PersonService: PersonService) {}
+  personList: Array<object> = this.personService
+    .getPersons()
+    .sort((a: any, b: any) => a.id - b.id);
+  constructor(public router: Router, private personService: PersonService) {}
 
   ngOnInit() {}
 
-  return() {
+  navigateBack() {
     this.router.navigate(["todo"]);
   }
 
   //--------- CRUD Operations -----------
 
   addPersonFromService() {
-    this.PersonService.addPerson({
-      id: this.personList.length + 1,
+    this.personService.addPerson({
+      id: this.personService.nextId,
```

Comment:
Since the ID is internal (you are not setting it). 
It should be handled only in the personService. 
You can set the ID to null here (only for placehodler) and the addPerson method will set the ID. 
Or you can change the addPerson to get the exact attributes